platform/SharedPtr.h

```diff
@@ -285,4 +287,8 @@ bool operator!= (U lhs, const SharedPtr<T> &rhs)
     return ((T *) lhs != rhs.get());
 }
 
+} /* namespace mbed */
+
+using mbed::SharedPtr;
```

Comment:
Along with addition of namespace and backward compatibility, we need to make sure we do not corrupt the global namespace hence a small change needed:

``using` should be under `#ifndef MBED_NO_GLOBAL_USING_DIRECTIVE` macro.
`
https://github.com/ARMmbed/mbed-os/pull/7760/files#diff-3716425795ebc1692d5a64771fa4f35eR169